lib/adapter.js

```diff
@@ -784,12 +784,9 @@ AbstractPouchDB.prototype.registerDependentDatabase =
     doc.dependentDbs[dependentDb] = true;
     return doc;
   }
-  upsert(this, '_local/_pouch_dependentDbs', diffFun, function (err) {
-    if (err) {
-      return callback(err);
-    }
-    return callback(null, {db: depDB});
-  });
+  upsert(this, '_local/_pouch_dependentDbs', diffFun)
+    .then(callback.bind(this, null, {db: depDB}))
```

Comment:
`.bind` tends to be bad for performance, the callback doesn't need `this`, and I believe it's shorter to just have a regular function here